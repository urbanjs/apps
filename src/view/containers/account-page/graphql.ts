import { gql } from 'react-apollo';

export const updateUserMutation = gql`  
mutation updateUserPersonalInformation($userId: ID!, $data: UserPersonalInformationInput!) {
  updateUserPersonalInformation(userId: $userId, data: $data) {
    id
    firstName
    lastName
    birthDate
    phoneNumber
    birthPlace
    socialSecurityNumber
    taxNumber
    mothersMaidenName
  }
}`;

export const userQuery = gql`
query {
  user {
    id
    email
    displayName
    avatar
    personalInformation {
      id
      firstName
      lastName
      birthDate
      phoneNumber
      birthPlace
      socialSecurityNumber
      taxNumber
      mothersMaidenName
    }
  }
}`;
