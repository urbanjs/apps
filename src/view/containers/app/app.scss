.zv-app {
  $sidebar_width: 4em;
  $footer_height: 3.3em;

  & > * {
    transition: all .4s ease-in;
  }

  &.zv-sidebar-hidden {
    .zv-sidebar {
      left: -$sidebar_width;
    }

    .zv-content-wrapper, .zv-footer-wrapper {
      margin-left: 0;
    }
  }

  .zv-sidebar {
    width: $sidebar_width;
  }

  .zv-content-wrapper {
    margin-left: $sidebar_width;
    margin-bottom: -$footer_height;
    min-height: 100%;

    &:after {
      content: "";
      display: block;
      height: $footer_height;
    }
  }

  .zv-footer-wrapper {
    margin-left: $sidebar_width;
    height: $footer_height;
    overflow: hidden;
  }
}
