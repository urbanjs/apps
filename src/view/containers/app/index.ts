export { AppWitHOCs as App } from './app';
