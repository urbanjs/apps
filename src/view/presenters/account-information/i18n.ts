export default {
  edit: 'common.action.edit',
  title: 'component.accountInformation.title'
};
