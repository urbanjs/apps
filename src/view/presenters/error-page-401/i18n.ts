export default {
  'error': 'common.error',
  '401': 'common.error.401'
};
