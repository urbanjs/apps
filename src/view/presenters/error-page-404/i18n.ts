export default {
  'error': 'common.error',
  '404': 'common.error.404'
};
