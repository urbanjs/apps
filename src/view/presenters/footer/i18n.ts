export default {
  'copyright': 'component.footer.copyright'
};
