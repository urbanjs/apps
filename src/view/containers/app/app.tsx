import * as React from 'react';
import { ActionCreator, connect, Dispatch } from 'react-redux';
import { Route, withRouter, RouteComponentProps, Switch } from 'react-router-dom';
import { QueryProps as ApolloQueryProps, graphql } from 'react-apollo';
import { PATH_APP, PATH_APP_ACCOUNT } from '../../../constants';
import { track } from '../../../decorators';
import { RootState } from '../../../state/reducers';
import { setLocale } from '../../../state/actions';
import { Sidebar, Navbar, Footer, Loader, ErrorPage404 } from '../../presenters';
import { AccountPage } from '../account-page';
import { Feature } from '../../../modules/authorization/types';
import { userQuery } from './graphql';
import './app.css';

export type OwnProps = {};

export type UserQueryProps = {
  data: ApolloQueryProps & {
    user?: {
      id: string;
      email: string;
      displayName: string;
      avatar: string;
      subscription: {
        features: Feature[]
      }
    }
  }
};

export type StateProps = {
  isLoading: boolean;
  currentLocale: string;
  locales: string[];
  serverOrigin: string;
  appOrigin: string;
};

export type DispatchProps = {
  setLocale: ActionCreator<object>;
};

export type AppProps =
  StateProps
  & DispatchProps
  & OwnProps
  & RouteComponentProps<null>
  & UserQueryProps;

export type State = { isSidebarCollapsed: boolean };

export class App extends React.Component<AppProps, State> {
  props: AppProps;
  state: State = {isSidebarCollapsed: false};

  @track()
  render() {
    if (this.props.isLoading) {
      return <Loader/>;
    }

    const allowedFeatures: Feature[] = this.props.data.user
      ? this.props.data.user.subscription.features
      : [];

    return (
      <div className={`zv-app h-100 ${this.state.isSidebarCollapsed ? 'zv-sidebar-hidden' : ''}`}>
        <aside className="zv-sidebar fixed-top h-100">
          <Sidebar allowedFeatures={allowedFeatures}/>
        </aside>

        <div className="zv-content-wrapper">
          <Navbar
            appOrigin={this.props.appOrigin}
            serverOrigin={this.props.serverOrigin}
            allowedFeatures={allowedFeatures}
            notifications={Array(100)}
            onCollapse={() => this.setState({isSidebarCollapsed: !this.state.isSidebarCollapsed})}
            user={this.props.data.user && {
              displayName: this.props.data.user.displayName,
              avatar: this.props.data.user.avatar,
              email: this.props.data.user.email
            }}
          />

          <Switch>
            <Route
              path={PATH_APP_ACCOUNT}
              render={() => <AccountPage/>}
            />

            <Route
              path={PATH_APP}
              exact={true}
            />

            <Route
              path="**"
              render={() => <ErrorPage404/>}
            />
          </Switch>
        </div>

        <footer className="zv-footer-wrapper">
          <Footer
            currentLocale={this.props.currentLocale}
            locales={this.props.locales}
            onLocaleChange={(locale: string) => this.props.setLocale(locale)}
          />
        </footer>
      </div>
    );
  }
}

export const AppWitHOCs =
  graphql<UserQueryProps, OwnProps>(userQuery)(
    withRouter<OwnProps>(
      connect<StateProps, DispatchProps>(
        (state: RootState): StateProps => ({
          currentLocale: state.i18n.locale,
          locales: state.i18n.availableLocales,
          isLoading: state.loader.isLoading,
          serverOrigin: state.runtime.variables.serverOrigin,
          appOrigin: state.runtime.variables.appOrigin
        }),
        (dispatch: Dispatch<RootState>): DispatchProps => ({
          setLocale: (locale: string) => dispatch(setLocale({locale}))
        })
      )(App)));
