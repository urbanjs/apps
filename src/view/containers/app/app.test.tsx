import 'reflect-metadata';
import { Container } from 'inversify';
import { initalizeContainer } from '../../../apps/utils/container';
import { authorizationModule } from '../../../modules/authorization';
import { routeModule } from '../../../modules/route';
import { logModule } from '../../../modules/log';

const container = new Container({defaultScope: 'Singleton'});
initalizeContainer(container, {lazyInject: true});

container.load(logModule);
container.load(authorizationModule);
container.load(routeModule);

import 'requestanimationframe';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import * as renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider as ApolloProvider } from 'react-apollo/test-utils';
import { Provider as StoreProvider, Store } from 'react-redux';
import { AppWitHOCs as App, OwnProps } from './app';
import { userQuery } from './graphql';
import { translations } from '../../../i18n';
import { createStore, RootState } from '../../../state';
import { PATH_APP } from '../../../constants';

describe('App', () => {
  let store: Store<RootState>;
  let apolloMocks: { request: { query: string }, result: object }[];
  let currentPath: string;
  let i18nMessages: { [key: string]: string };

  beforeEach(() => {
    store = createStore(
      {
        runtime: {
          variables: {
            appOrigin: 'appOrigin',
            serverOrigin: 'serverOrigin'
          }
        }
      },
      {platform: 'browser'}
    );

    apolloMocks = [
      {request: {query: userQuery}, result: {data: {user: null}}}
    ];

    currentPath = PATH_APP;

    i18nMessages = translations.en;
  });

  const renderComponent = (props?: OwnProps) => {
    const component = (
      <MemoryRouter initialEntries={[currentPath]}>
        <StoreProvider store={store}>
          <ApolloProvider mocks={apolloMocks}>
            <IntlProvider locale="en" messages={i18nMessages}>
              <App {...props}/>
            </IntlProvider>
          </ApolloProvider>
        </StoreProvider>
      </MemoryRouter>
    );

    return renderer.create(component);
  };

  it('renders without crashing', () => {
    const component = renderComponent();

    expect(component.toJSON()).toMatchSnapshot();
  });
});
