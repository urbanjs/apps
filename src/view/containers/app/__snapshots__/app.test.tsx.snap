// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="zv-app h-100 "
>
  <aside
    className="zv-sidebar fixed-top h-100"
  >
    <div
      className="zv-sidebar bg-faded h-100"
    >
      <a
        className="zv-logo-link btn btn-link text-muted d-block mx-auto mt-4 mb-4 active"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa fa-2x"
        />
      </a>
    </div>
  </aside>
  <div
    className="zv-content-wrapper"
  >
    <div
      className="zv-navbar p-4 d-flex justify-content-end bg-faded dropdown align-items-center "
    >
      <a
        className="btn btn-link text-muted mr-auto"
        onClick={[Function]}
      >
        <i
          className="fa fa-bars"
        />
      </a>
      <a
        className="btn btn-link text-muted"
        onClick={[Function]}
      >
        <i
          className="fa fa-2x fa-sign-in"
        />
      </a>
      <div
        className="dropdown-menu dropdown-menu-right p-4 mt-0"
        onClick={[Function]}
      >
        <a
          className="btn btn-primary w-100"
          href="serverOrigin/auth/facebook?redirect_uri=appOrigin%2F"
        >
          <span>
            Sign in
          </span>
        </a>
      </div>
    </div>
  </div>
  <footer
    className="zv-footer-wrapper"
  >
    <div
      className="zv-footer bg-faded d-flex justify-content-between"
    >
      <div
        className="p-4 text-muted"
      >
          ©  
        <span>
          Copyright
        </span>
         
        2017
      </div>
      <div
        className="p-2"
      >
        <select
          className="form-control"
          onChange={[Function]}
          value="en"
        >
          <option>
            en
          </option>
          <option>
            hu
          </option>
        </select>
      </div>
    </div>
  </footer>
</div>
`;
